### DISCUSSION TITLE
Contribute emscripten tool to SCons?

### DISCUSSION THREAD

**Discussion by bdbaddog** [emscripten-core/emscripten#19355] [General]
Greetings from the SCons project.

Would you consider contributing (PR) your scons tool to our project.
We'd like to include it and make it part of our regular releases.

-Bil
SCons Project Co-Manager

**Comment by sbc100** [comment]
If there is some way we can take what we have in the `tools/scons/site_scons/` and move it upstream that would be great.  Could you take a look at what we have there and see what you think?

Sadly I think it would be a long time before we would be able to delete the downstream code here since we want to be able to run on linux distributions which might have older scons for quite a while...

**Comment by bdbaddog** [comment]
@sbc100 - I'll take a look. 
Looks like your code is MIT licensed (+ another), but MIT is the license we use for SCons so that shouldn't be an issue.

For a PR we require doc updates/additions, tests, and updates to our CHANGES.txt/RELEASE.txt

Hopefully we can upstream you fairly quickly.
Do you have some cycles to assist with the above items?

**Reply by sbc100** [reply]
IIUC there are very few users of our scons plugin, so I don't think the core emscripten developers would be able to prioritize upstreaming or working on the scons plugin, but we would be happy to review any PRs that get sent our way.
